api/src/bench.rs

```diff
@@ -85,12 +78,39 @@ pub trait Bench {
     /// Get a manifest of the builtin actors
     fn builtin_actors_manifest(&self) -> BTreeMap<Cid, vm_api::builtin::Type>;
 
+    /// Get a map of all address -> actor mappings in the state tree
+    fn actor_states(&self) -> BTreeMap<Address, ActorState>;
+
+    /// Get the VM's current epoch
+    fn epoch(&self) -> ChainEpoch;
+
+    /// Set the VM's current epoch
+    fn set_epoch(&mut self, epoch: ChainEpoch);
+
     /// Get the current circulating supply
     fn circulating_supply(&self) -> TokenAmount;
 
     /// Set the current circulating supply
     fn set_circulating_supply(&mut self, amount: TokenAmount);
 
-    /// Get a map of all address -> actor mappings in the state tree
-    fn actor_states(&self) -> BTreeMap<Address, ActorState>;
+    /// Get the current base fee
+    fn base_fee(&self) -> TokenAmount;
+
+    /// Set the current base fee
+    fn set_base_fee(&mut self, amount: TokenAmount);
+
+    /// Get the current timestamp
+    fn timestamp(&self) -> u64;
+
+    /// Set the current timestamp
+    fn set_timestamp(&mut self, timestamp: u64);
+
+    /// Get the initial state root of the block
+    fn initial_state_root(&self) -> Cid;
+
+    /// Set the initial state root of the block
+    fn set_initial_state_root(&mut self, state_root: Cid);
```

Comment:
I think we can omit this (in VM API). It's not exposed to the actors (there's a syscall called state root, but it's the actor's root, not the whole block's root). It's conceptually incompatible to abstract of the state tree collection of actors but also talk about this root that implies the specific HAMT etc.